Tests/ApolloCodegenTests/CodeGenIR/IRRootFieldBuilderTests.swift

```diff
@@ -4090,6 +4090,151 @@ class IRRootFieldBuilderTests: XCTestCase {
       .to(shallowlyMatch(predator_asWarmBlooded_height_expected))
   }
 
+  func test__mergedSelections__givenEntityField_mergedSelectionsFromNamedFragment_intoNestedEntities_withDepthOf4_nestedEntitiesHaveMergedSelectionsAndSources() async throws {
+    // given
+    schemaSDL = """
+    type Query {
+      root: Root
+    }
+
+    type Root {      
+      foo: Foo
+    }
+
+    type Foo {
+      a: Int
+      b: Int
+      bar: Bar
+    }
+    
+    type Bar {
+      a: Int
+      b: Int      
+      foo: Foo   
+    }
+    """
+
+    document = """
+    query TestOperation {
+      root {
+        ...FooFragment
+        foo {
+          a
+          bar {
+            a
+            foo {
+              a
+              bar {
+                a
+              }
+            }
+          }
+        }
+      }
+    }
+
+    fragment FooFragment on Root {
+      foo {
+        b
+        bar {
+          b
+          foo {
+            b
+            bar {
+              b              
+            }
+          }
+        }
+      }
+    }    
+    """
+
+    // when
+    try await buildSubjectRootField()
+
+    let Object_Root = try XCTUnwrap(schema[object: "Root"])
```

Comment:
There is a warning that this variable is never used - is that intentional or highlighting something missed in the test?

Reply:
Oh you're right. For some reason the warning did not show up for me in Xcode? Fixed!